### About

This is the notebook for preparing data for hero_prediction model.


In [1]:
import pandas as pd

In [2]:
df = pd.read_csv('cleanData.csv')

In [3]:
champions = df.championid.unique().tolist()
championsNumber = len(champions)
print(championsNumber)

136


#### NOTE:

The most recent patch version appears in the datasets is 7.10 (ie around May 17
2017). Number of champions play can choose is 136.


In [4]:
#creating columns name
team1champions = [0] * (championsNumber)
for i in range(len(team1champions)):
    team1champions[i] = "t1_"+str(champions[i])  
team2champions = [0] * (championsNumber)
for i in range(len(team2champions)):
    team2champions[i] = "t2_"+str(champions[i])  
resultantCols = team1champions
resultantCols.extend(team2champions)
resultantCols.append('team1Win')
len(resultantCols)

273

In [5]:
rows_list = []
tempRow = [0] * (2*championsNumber)
for idx, x in df['championid'].iteritems():
    
    if((idx+1)%10!=0 and (idx+1)%10<6):#team1 selection
        i = champions.index(x)
        tempRow[i] = 1.0
    else:#team2 selection
        i = champions.index(x)
        tempRow[2*i+1] = 1.0
    
    
    #new match, reset the tempRow    
    if((idx+1)%10==0):         
        winOrNot = df.get_value(idx, 'win')
        if(winOrNot==1):   #ie team 2 win
            winOrNot = -1.0 #ie team 1 lose
        else:
            winOrNot = 1.0
        
        tempRow.append(winOrNot)
        
        #we need champions with two times
        tempD = dict(zip(resultantCols ,tempRow))
        rows_list.append(tempD)
        tempRow = [0] * (2*championsNumber)   
        


C:\Users\tim\Anaconda3\envs\lol-predict\lib\site-packages\ipykernel_launcher.py:15: FutureWarning: get_value is deprecated and will be removed in a future release. Please use .at[] or .iat[] accessors instead
  from ipykernel import kernelapp as app


In [6]:
heroSelect = pd.DataFrame(rows_list)

#sorting team1 heros after team2 and team1Win as the first column
heroSelect.reindex_axis(sorted(heroSelect.columns), axis=1)
cols = heroSelect.columns.tolist()
cols.insert(0, cols.pop(cols.index('team1Win')))
heroSelect = heroSelect.reindex(columns=cols)

C:\Users\tim\Anaconda3\envs\lol-predict\lib\site-packages\ipykernel_launcher.py:4: FutureWarning: '.reindex_axis' is deprecated and will be removed in a future version. Use '.reindex' instead.
  after removing the cwd from sys.path.


In [7]:
heroSelect.to_csv('heroSelect.csv')
heroSelect.head(20)

team1Win  t1_1  t1_10  t1_101  t1_102  t1_103  t1_104  t1_105  t1_106  \
0       -1.0   0.0    0.0     0.0     0.0     0.0     0.0     0.0     0.0   
1       -1.0   0.0    0.0     0.0     0.0     0.0     0.0     0.0     0.0   
2       -1.0   0.0    0.0     0.0     0.0     0.0     0.0     0.0     0.0   
3       -1.0   0.0    0.0     0.0     0.0     0.0     0.0     1.0     0.0   
4       -1.0   0.0    0.0     0.0     0.0     0.0     1.0     0.0     0.0   
5        1.0   0.0    0.0     0.0     0.0     1.0     0.0     0.0     0.0   
6       -1.0   0.0    1.0     0.0     0.0     0.0     0.0     0.0     0.0   
7       -1.0   0.0    0.0     0.0     0.0     0.0     1.0     0.0     0.0   
8       -1.0   0.0    0.0     0.0     0.0     0.0     0.0     0.0     0.0   
9        1.0   0.0    0.0     0.0     0.0     0.0     0.0     0.0     0.0   
10      -1.0   0.0    0.0     0.0     1.0     0.0     0.0     1.0     0.0   
11      -1.0   0.0    0.0     0.0     1.0     0.0     0.0     1.0     0.0   
12       1.0   0.0    0.0     0.0     0.0     0.0     1.0     0.0     0.0   
13       1.0   0.0    0.0     0.0     0.0     0.0     1.0     1.0     0.0   
14       1.0   0.0    0.0     0.0     0.0     0.0     0.0     0.0     0.0   
15       1.0   0.0    0.0     0.0     0.0     0.0     0.0     0.0     0.0   
16       1.0   0.0    0.0     0.0     0.0     0.0     1.0     0.0     0.0   
17      -1.0   0.0    0.0     0.0     0.0     0.0     0.0     0.0     0.0   
18       1.0   0.0    0.0     0.0     0.0     1.0     1.0     0.0     0.0   
19       1.0   0.0    0.0     0.0     0.0     0.0     0.0     0.0     0.0   

    t1_107  ...    t2_85  t2_86  t2_89  t2_9  t2_90  t2_91  t2_92  t2_96  \
0      0.0  ...      0.0    0.0      0     0      0      0    0.0      0   
1      0.0  ...      0.0    0.0      0     0      0      0    0.0      0   
2      0.0  ...      0.0    0.0      0     0      0      0    0.0      0   
3      0.0  ...      0.0    0.0      0     0      0      0    0.0      0   
4      0.0  ...      0.0    0.0      0     0      0      0    0.0      0   
5      0.0  ...      0.0    0.0      0     0      0      0    0.0      0   
6      0.0  ...      0.0    0.0      0     0      0      0    0.0      0   
7      0.0  ...      0.0    0.0      0     0      0      0    0.0      0   
8      0.0  ...      0.0    0.0      0     0      0      0    0.0      0   
9      0.0  ...      0.0    0.0      0     0      0      0    0.0      0   
10     0.0  ...      0.0    0.0      0     0      0      0    0.0      0   
11     0.0  ...      0.0    0.0      0     0      0      0    0.0      0   
12     1.0  ...      0.0    0.0      0     0      0      0    0.0      0   
13     0.0  ...      0.0    0.0      0     0      0      0    0.0      0   
14     0.0  ...      0.0    0.0      0     0      0      0    0.0      0   
15     0.0  ...      0.0    0.0      0     0      0      0    0.0      0   
16     0.0  ...      0.0    0.0      0     0      0      0    0.0      0   
17     0.0  ...      0.0    0.0      0     0      0      0    0.0      0   
18     0.0  ...      0.0    0.0      0     0      0      0    0.0      0   
19     0.0  ...      0.0    0.0      0     0      0      0    0.0      0   

    t2_98  t2_99  
0       0      0  
1       0      0  
2       0      0  
3       0      0  
4       0      0  
5       0      0  
6       0      0  
7       0      0  
8       0      0  
9       0      0  
10      0      0  
11      0      0  
12      0      0  
13      0      0  
14      0      0  
15      0      0  
16      0      0  
17      0      0  
18      0      0  
19      0      0  

[20 rows x 273 columns]